test/integration/test_payroll_process_scenarios_2/payroll_index_payment.integration.test.js

```diff
@@ -0,0 +1,105 @@
+/* global expect, agent */
+const helpers = require('../helpers');
+
+describe(`Tests to verify that enabling "Payment by Index"
+  correctly updates the BHIMA system for index-based payments`, () => {
+  const defaultEnterpriseId = 1;
+
+  // ---------------------------------------------------------
+  // This test verifies that when the index payment system is
+  // enabled, the payroll payslip report is generated correctly
+  // and that the correct report template is used.
+  it('should generate payroll payslip and verify that rubricsIndexes (otherProfits) contain elements', async () => {
+    const paramPayslip = {
+      currency : '2',
+      employees : [
+        '8FAC0B70C12B423FB575324751448F4F',
+        '0236AF7EBDF449B6A4F0309EC94BC92A',
+      ],
+      idPeriod : '19',
+      lang : 'fr',
+      renderer : 'json', // JSON to easily inspect the data
+    };
+
+    try {
+      // Step 1: Check the enterprise configuration
+      const enterpriseRes = await agent.get(`/enterprises/${defaultEnterpriseId}`);
+      expect(enterpriseRes).to.have.status(200);
+      expect(enterpriseRes.body.settings.enable_index_payment_system).to.equal(1);
+
+      // Step 2: Call the API to generate the report
+      const reportRes = await agent.get(`/reports/payroll/payslip`).query(paramPayslip);
+      expect(reportRes).to.have.status(200);
+      expect(reportRes.body).to.be.an('object');
+      expect(reportRes.body.dataEmployees).to.be.an('array');
+
+      // Step 3: Verify rubricsIndexes via otherProfits
+      reportRes.body.dataEmployees.forEach(employee => {
+        expect(employee).to.have.property('otherProfits');
+
+        // Test only if the array contains elements
+        if (employee.otherProfits.length > 0) {
+          employee.otherProfits.forEach(profit => {
+            expect(profit).to.have.property('value');
```

Comment:
You can simplify both these assertions to be :

```js
expect(profit).to.have.all.keys('label', 'value');
```

I believe this will throw an error if there are more keys than just those two.   If we have more keys that we return, let's check for those here.